extensions/amp-viewer-integration/0.1/findtext.js

```diff
@@ -118,65 +215,76 @@ export function canonicalizeString(s) {
  * @return {?Array<!TextRangeDef>}
  */
 export function findSentences(win, node, sentences) {
+  sentences = canonicalizeSentences(sentences);
+  if (sentences.length <= 0) {
+    return null;
+  }
   const scanner = new TextScanner(win, node);
-  const matches = [];
-  for (let senIdx = 0; senIdx < sentences.length; senIdx++) {
-    const sen = canonicalizeString(sentences[senIdx]);
-    if (!sen) {
+  // Creates a circular buffer with capacity = max(size of sentence).
+  /** @type {!CircularBuffer<!TextPosAndIdxDef>} */
+  const buf = new CircularBuffer(
+    sentences.map(sen => sen.length).reduce((x, y) => Math.max(x, y))
```

Comment:
nit: Can you use `Math.max` directly as argument to reduce, rather than wrapping it in an arrow func?

Reply:
Yes, I initially tried it and it turned out it does not work :)
I added a comment.

Reply:
Makes sense. Today I learned :)